acvm-repo/acir/src/circuit/brillig.rs

```diff
@@ -46,3 +49,66 @@ impl std::fmt::Display for BrilligFunctionId {
         write!(f, "{}", self.0)
     }
 }
+
+/// Procedures are a set of complex operations that are common in the noir language.
+/// Extracting them to reusable procedures allows us to reduce the size of the generated Brillig.
+/// Procedures receive their arguments on scratch space to avoid stack dumping&restoring.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, PartialOrd, Ord, Deserialize, Serialize)]
+pub enum ProcedureId {
```

Comment:
Can you move this up into the compiler? Procedures are Noir-only constructs and so shouldn't be in the ACIR crate.

Reply:
Yeah I was unsure on this change. I did this due to `ProcedureId` being added to the `DebugInfo` which is in `noirc_errors`. I'll make a separate ID mapping for `ProcedureId`.

Reply:
I'll look at making a separate ID mapping for procedure debug info.

Reply:
I have a new `ProcedureDebugId` in `noirc_errors` that we convert into when creating the brillig procedures debug data in ACIR and then convert back into a `ProcedureId` in the flamegraph.